### PR TITLE
Chinese Simplified: Developer Tutorials

### PR DESCRIPTION
**Pull request by wackerow**
## Description
Updates Chinese Simplified (`zh`) "Developer Tutorials" content with latest from Crowdin

### CODE REVIEW

File: src/content/translations/zh/developers/tutorials/waffle-dynamic-mocking-and-testing-calls/index.md [hunk 1]
```diff
@@ -280,11 +280,11 @@ it("checks if contract called balanceOf with certain wallet on the ERC20 token",
 
 让我们检查测试是否正确：
 
-![三次测试通过](../../../../../developers/tutorials/waffle-dynamic-mocking-and-testing-calls/test-three.png)
+![三次测试通过](test-three.png)
```
Comment (minimalsm) [inline src/content/translations/zh/developers/tutorials/waffle-dynamic-mocking-and-testing-calls/index.md:283]:
```suggestion
![三次测试通过](../../../../../developers/tutorials/waffle-dynamic-mocking-and-testing-calls/test-three.png)
```

File: src/content/translations/zh/developers/tutorials/waffle-dynamic-mocking-and-testing-calls/index.md [hunk 2]
```diff
@@ -219,14 +219,14 @@ it("returns false if the wallet has less than 1000000 tokens", async () => {
...
-1. 将我们的模拟 ERC20 合约设置为始终返回 99999 个通证的余额。
-2. 检查`contract.check()`方法是否返回`false`。
+1. 将我们的模拟 ERC20 合约设置为始终返回 99999 个代币的余额。
+2. 检查 `contract.check()` 方法是否返回 `false`。
 
-我们已经准备好启动这个野兽了：
+我们已经准备好启动这个家伙了：
 
-![一次测试通过](../../../../../developers/tutorials/waffle-dynamic-mocking-and-testing-calls/test-one.png)
+![一次测试通过](test-one.png)
```
Comment (minimalsm) [inline src/content/translations/zh/developers/tutorials/waffle-dynamic-mocking-and-testing-calls/index.md:227]:
```suggestion
![一次测试通过](../../../../../developers/tutorials/waffle-dynamic-mocking-and-testing-calls/test-one.png)
```

File: src/content/translations/zh/developers/tutorials/waffle-dynamic-mocking-and-testing-calls/index.md [hunk 3]
```diff
@@ -250,13 +250,13 @@ it("returns true if the wallet has at least 1000001 tokens", async () => {
 
 您运行测试...
 
-![两次测试通过](../../../../../developers/tutorials/waffle-dynamic-mocking-and-testing-calls/test-two.png)
+![两次测试通过](test-two.png)
```
Comment (minimalsm) [inline src/content/translations/zh/developers/tutorials/waffle-dynamic-mocking-and-testing-calls/index.md:253]:
```suggestion
![两次测试通过](../../../../../developers/tutorials/waffle-dynamic-mocking-and-testing-calls/test-two.png)
```

File: src/content/translations/zh/developers/tutorials/using-websockets/index.md [hunk 4]
```diff
@@ -45,7 +46,7 @@ $ wscat -c wss://eth-mainnet.ws.alchemyapi.io/ws/demo
 
 首先，使用应用的 WebSocket URL 打开 WebSocket 您可以在[您的仪表板](https://dashboard.alchemyapi.io/)中打开应用的页面并点击“查看密钥”来找到您的应用的 WebSocket URL。 请注意，您的应用的 WebSocket URL 与其 HTTP 请求的 URL 不同，但两者都可以通过点击“查看密钥” 找到。
 
-![在您的Alchemy仪表板中的何处可找到您的WebSocket URL](../../../../../developers/tutorials/using-websockets/use-websockets.gif)
+![在您的Alchemy仪表板中的何处可找到您的WebSocket URL](../../../../developers/tutorials/using-websockets/use-websockets.gif)
```
Comment (minimalsm) [inline src/content/translations/zh/developers/tutorials/using-websockets/index.md:49]:
```suggestion
![在您的Alchemy仪表板中的何处可找到您的WebSocket URL](../../../../../developers/tutorials/using-websockets/use-websockets.gif)
```

File: src/content/translations/zh/developers/tutorials/transfers-and-approval-of-erc-20-tokens-from-a-solidity-smart-contract/index.md [hunk 5]
```diff
@@ -185,7 +185,7 @@ XPath: /pre[3]/code
 
 在购买成功的情况下，我们应会在交易中看到两个事件：通证`Transfer`和`Bought`事件。
 
-![交易中的两个事件：Transfer和Bought](../../../../../developers/tutorials/transfers-and-approval-of-erc-20-tokens-from-a-solidity-smart-contract/transfer-and-bought-events.png)
+![交易中的两个事件：Transfer和Bought](../../../../developers/tutorials/transfers-and-approval-of-erc-20-tokens-from-a-solidity-smart-contract/transfer-and-bought-events.png)
```
Comment (minimalsm) [inline src/content/translations/zh/developers/tutorials/transfers-and-approval-of-erc-20-tokens-from-a-solidity-smart-contract/index.md:188]:
```suggestion
![交易中的两个事件：Transfer和Bought](../../../../../developers/tutorials/transfers-and-approval-of-erc-20-tokens-from-a-solidity-smart-contract/transfer-and-bought-events.png)
```

File: src/content/translations/zh/developers/tutorials/the-graph-fixing-web3-data-querying/index.md [hunk 6]
```diff
@@ -72,68 +72,47 @@ GameContract.events.BetPlaced({
...
+![一个不简单的查询](../../../../developers/tutorials/the-graph-fixing-web3-data-querying/one-does-not-simply-query.jpg)
 
 您可以看到这并不是最优的：
 
 - 不适用于已经部署的合约。
 - 存储这些值需要额外的 gas 成本。
 - 需要另一个调用来获取以太坊节点的数据。
 
-![是不够好的](../../../../../developers/tutorials/the-graph-fixing-web3-data-querying/not-good-enough.jpg)
+![是不够好的](../../../../developers/tutorials/the-graph-fixing-web3-data-querying/not-good-enough.jpg)
```
Comment (minimalsm) [inline src/content/translations/zh/developers/tutorials/the-graph-fixing-web3-data-querying/index.md:83]:
```suggestion
![是不够好的](../../../../../developers/tutorials/the-graph-fixing-web3-data-querying/not-good-enough.jpg)
```

File: src/content/translations/zh/developers/tutorials/the-graph-fixing-web3-data-querying/index.md [hunk 7]
```diff
@@ -72,68 +72,47 @@ GameContract.events.BetPlaced({
 
 现在，对于我们的简单示例来说，这在某种程度上还是不错的。 但是假设我们现在只想显示当前玩家输/赢的赌注金额。 嗯，我们运气不好，您最好部署一份新的合约来存储这些值，并将它们提取出来。 现在想象一下一个更复杂的智能合约和去中心化应用，事情可能很快就会变得一团糟。
 
-![一个不简单的查询](../../../../../developers/tutorials/the-graph-fixing-web3-data-querying/one-does-not-simply-query.jpg)
+![一个不简单的查询](../../../../developers/tutorials/the-graph-fixing-web3-data-querying/one-does-not-simply-query.jpg)
```
Comment (minimalsm) [inline src/content/translations/zh/developers/tutorials/the-graph-fixing-web3-data-querying/index.md:75]:
```suggestion
![一个不简单的查询](../../../../../developers/tutorials/the-graph-fixing-web3-data-querying/one-does-not-simply-query.jpg)
```

**Review by minimalsm — CHANGES_REQUESTED** [review]
:]

File: src/content/translations/zh/developers/tutorials/the-graph-fixing-web3-data-querying/index.md [hunk 8]
```diff
@@ -72,68 +72,47 @@ GameContract.events.BetPlaced({
...
-为方便起见，您可能还想使用像 mustache 这样的模板工具。 然后您可以创建一个 subgraph.template.yaml，并根据最新的部署插入相应的地址。 有关更高级的示例设置，请参阅这个[Aave subgraph repo](https://github.com/aave/aave-protocol/tree/master/thegraph)示例。
-
-完整的文档可以在这里看到：https://thegraph.com/docs/define-a-subgraph#the-subgraph-manifest。
+为方便起见，您可能还想使用像 mustache 这样的模板工具。 然后您可以创建一个 subgraph.template.yaml，并根据最新的部署插入相应的地址。 有关更高级的示例设置，请参阅这个 [Aave subgraph repo](https://github.com/aave/aave-protocol/tree/master/thegraph) 示例。
 
 ```yaml
 specVersion: 0.0.1
 description: Placing Bets on Ethereum
-repository: - GitHub link -
+repository: - Github link -
```
Comment (wackerow) [inline src/content/translations/zh/developers/tutorials/the-graph-fixing-web3-data-querying/index.md:115]:
```suggestion
repository: - GitHub link -
```

File: src/content/translations/zh/developers/tutorials/downsizing-contracts-to-fight-the-contract-size-limit/index.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+  - "truffle"
+skill: 中级难度
+published: 2020-06-26
+source: soliditydeveloper.com
+sourceUrl: https://soliditydeveloper.com/max-contract-size
+---
+
+## 为什么会有限制？ {#why-is-there-a-limit}
+
+在 2016 年 [11 月 22 日](https://blog. ethereum. org/2016/11/18/hard-fork-no-4-spurious-dragon/)， Spilious Dragon 硬分叉引入了 [EIP-170](https://eips. ethereum. org/EIPS/eip-170)，增加了 24.576 kb 的智能合约大小限制。 对于作为 Solidity 开发者的您来说，这意味着当您向合约中添加越来越多的功能时，在某个时候您会达到极限，并且在部署时会看到错误：
```
Comment (corwintines) [inline src/content/translations/zh/developers/tutorials/downsizing-contracts-to-fight-the-contract-size-limit/index.md:20]:
```suggestion
在 2016 年 [11 月 22 日](https://blog.ethereum.org/2016/11/18/hard-fork-no-4-spurious-dragon/)， Spilious Dragon 硬分叉引入了 [EIP-170](https://eips.ethereum.org/EIPS/eip-170)，增加了 24.576 kb 的智能合约大小限制。 对于作为 Solidity 开发者的您来说，这意味着当您向合约中添加越来越多的功能时，在某个时候您会达到极限，并且在部署时会看到错误：
```

**Review by corwintines — APPROVED** [review]

**Pull request merged**